src/OpenTelemetry.Extensions/Trace/PartialActivityProcessor.cs

```diff
@@ -0,0 +1,273 @@
+        AccessOpenTelemetryInternals(
+            out writeTraceDataMethod,
+            out logRecordConstructor,
+            out sdkLimitOptions);
+
+        this.exporterThread = new Thread(this.ExporterProc)
+        {
+            IsBackground = true, Name = $"OpenTelemetry-{nameof(PartialActivityProcessor)}",
+        };
+        this.exporterThread.Start();
+    }
+
+    /// <summary>
+    /// Gets for testing purposes.
+    /// </summary>
+    public IReadOnlyDictionary<ActivitySpanId, Activity> ActiveActivities => this.activeActivities;
+
+    /// <summary>
+    /// Gets for testing purposes.
+    /// </summary>
+    public IReadOnlyCollection<KeyValuePair<ActivitySpanId, Activity>> EndedActivities =>
+        this.endedActivities;
+
+    /// <inheritdoc />
+    public override void OnStart(Activity data)
+    {
+        if (data == null)
+        {
+            return;
+        }
+
+        var logRecord = GetLogRecord(data, this.GetHeartbeatLogRecordAttributes());
+        this.logExporter.Export(new Batch<LogRecord>(logRecord));
+        this.activeActivities[data.SpanId] = data;
+    }
+
+    /// <inheritdoc />
+    public override void OnEnd(Activity data)
+    {
+        if (data == null)
+        {
+            return;
+        }
+
+        var logRecord = GetLogRecord(data, GetStopLogRecordAttributes());
+        this.logExporter.Export(new Batch<LogRecord>(logRecord));
+        this.endedActivities.Enqueue(new KeyValuePair<ActivitySpanId, Activity>(data.SpanId, data));
+    }
+
+    /// <inheritdoc />
+    protected override bool OnShutdown(int timeoutMilliseconds)
+    {
+        try
+        {
+            this.shutdownTrigger.Set();
+        }
+        catch (ObjectDisposedException)
+        {
+            return false;
+        }
+
+        switch (timeoutMilliseconds)
+        {
+            case Timeout.Infinite:
+                this.exporterThread.Join();
+                return this.logExporter.Shutdown();
+            case 0:
+                return this.logExporter.Shutdown(0);
+        }
+
+        var sw = Stopwatch.StartNew();
+        this.exporterThread.Join(timeoutMilliseconds);
+        var timeout = timeoutMilliseconds - sw.ElapsedMilliseconds;
+        return this.logExporter.Shutdown((int)Math.Max(timeout, 0));
+    }
+
+    /// <inheritdoc />
+    protected override void Dispose(bool disposing)
+    {
+        base.Dispose(disposing);
+        this.shutdownTrigger.Dispose();
+    }
+
+    private static LogRecord GetLogRecord(
+        Activity data,
+        List<KeyValuePair<string, object?>> logRecordAttributesToBeAdded)
+    {
+        var buffer = new byte[750000];
+
+        var result = writeTraceDataMethod.Invoke(
+            null,
+            [buffer, 0, sdkLimitOptions, null!, new Batch<Activity>(data)]);
+        var writePosition = result as int? ?? 0; // Use a default value if null
+
+        var logRecord = (LogRecord)logRecordConstructor.Invoke(null);
```

Comment:
I think we should also set the `Severity` and `EventId`.

Reply:
What about `INFO` for `Severity`?

About `EventId`, it get a little complicated. In docs it says that it should be only set when used through `ILogger` - and this is not the case. From where/what would it value be derived from?

Reply:
> What about INFO for Severity?

Works for me.

> From where/what would it value be derived from?

This should be something which defines the name of the event. Maybe simply `span.state`.

Example: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/gen-ai/gen-ai-events.md